pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -400,6 +400,16 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 		}
 	}
 
+	if operConfig.Spec.Migration != nil {
```

Comment:
should we also check for `Migration.NetworkType`?
There is a chance someone want's to migrate to a different MTU and `NetworkType` can be unset.

Also, do we need to check that `Migration.NetworkType` actually changed to handle a scenario where the value is set to the current CNI or one that is not supported at all?

Reply:
+1

Or another version of this: should we check that the resources we are migrating to don't already exist?

Reply:
> should we also check for Migration.NetworkType?

I'll add that.

>Or another version of this: should we check that the resources we are migrating to don't already exist?

I use `apply.ApplyObject` to create the resources. So I don't think we need to check if one resource already exists.

Reply:
Not asking from the perspective of it failing or not. But if it exists, we can potentially overwrite something. Is that ok?

Reply:
The answer to this is that the user is not expected to make any changes during migration or within the time window where it decides whether the rollback has been successful or not. After that, rollback is no longer supported.
Also the user is not expected to apply the egress firewall CRD and add egress firewalls manually before migration.
I don't think that a check for existence would do any harm and probably make the migration more robust, but not a strong opinion on it. I am fine with it as it is since any information loss would be responsibility of the customer if the above requirements are clearly layed out in the migration documentation.

Reply:
>Also, do we need to check that Migration.NetworkType actually changed to handle a scenario where the value is set to the current CNI or one that is not supported at all?

I don't think it matters if the Migration.NetworkType is set to the current CNI. If for example current CNI is OVN and Migration.NetworkType is also OVN, it will attempt to migrate SDN resources to OVN. If this has been done already, that is okay, no change will be applied. If the SDN resources no longer exist, an error will be returned, and will disappear shortly after when migration finishes.

What value for CNI would not be supported? I suppose we can check like this:
```
if operConfig.Spec.Migration != nil && operConfig.Spec.Migration.NetworkType != "" 
&& (operConfig.Spec.Migration.NetworkType == "OpenshiftSDN" ||
operconfig.Spec.Migration.NetworkType == "OVNKubernetes") {
	migration := operConfig.Spec.Migration
	if migration.Features == nil || migration.Features.EgressFirewall {
		err = migrateEgressFirewallCRs(ctx, operConfig, r.client)
		if err != nil {
			return reconcile.Result{}, err
		}
	}
}

Reply:
Update: I'll address these changes in my own PR since they aren't critical. For now, let's get this merged.